data_scripts.csh

```diff
@@ -30,8 +31,24 @@ else if ($CONTINUE_RUN == TRUE) then
    set FILE = `head -n 1 ${data_scratch}/run/rpointer.atm_0001`
    set ATM_DATE_EXT = $FILE:r:e
    set ATM_DATE     = `echo $ATM_DATE_EXT | sed -e "s#-# #g"`
-   setenv data_year   `echo $ATM_DATE[1] | bc`
-   setenv data_month  `echo $ATM_DATE[2] | bc`
+   setenv d_year  `echo $ATM_DATE[1] | bc`
+   setenv d_month `echo $ATM_DATE[2] | bc`
+   # These must be defined using @ so that they are numbers (not strings)
+   # which can be used in math expressions.
+   # It appears that they are inherited by the calling scripts,
+   # even though they are not set with setenv.
+   @ data_year  = $d_year
+   @ data_month = $d_month
+
+   # If the assimilation ended on the first day of the month
+   # then the month assimilated is the previous.
+   if ($ATM_DATE[3] == '01' ) then
+      if ($data_month == 1) then
+         @ data_month = 12
```

Comment:
Do we have to decrement the year too?

Reply:
i can't figure out exactly how to post this since it's not on a changed line - but the comment in this file on line 38 where it says 'inherited by calling scripts' doesn't parse.  
'inherited' indicates for scripts called *by* this one.  if you mean the script calling this one sees the changes, then maybe 'changed in the calling scripts' parses ok.  or if you mean changes in the scripts called by this one, then you can use inherited and say called by this script.  a minor issue, yes, but it's confusing as to which direction in the calling/called stack of scripts you're trying to say these changes are made.